### PR TITLE
Fix/group name editor

### PR DESCRIPTION
**Pull request by Jungyubw**
Group Name issue fixed

### CODE REVIEW

**Review by afrold — COMMENTED** [review]
@Jungyubw please use material input.

**Review by afrold — CHANGES_REQUESTED** [review]
Please use material input.

**Comment by Jungyubw** [comment]
Okay wait for me.

**Comment by Jungyubw** [comment]
Done, let me know when you deploy the tools.

File: igamt-lite-client/app/views/edit/editMessages.html [hunk 1]
```diff
@@ -625,21 +625,37 @@
             </a>
             <ng-form class="form-inline">
                 <span class="label label-success label-icon">G</span>
+                <span ng-show="editableGrp!==node.obj.id" context-menu="segOption" ng-bind="getGroupNodeName(node.obj)" />
                 <span ng-show="editableGrp===node.obj.id">
-                    <select ng-show="editableGrp===node.obj.id" ng-model="position" ng-change="selectPos()"  style="width:15%" class="form-control input-sm">
```
Comment (afrold) [inline igamt-lite-client/app/views/edit/editMessages.html]: The ng-show is missing in the material version. Is that normal ? or by mistake ?

File: igamt-lite-client/app/views/edit/editMessages.html [hunk 2]
```diff
@@ -625,21 +625,37 @@
...
-                        <option ng-repeat="child in $root.segParent.children"  ng-selected="node.obj.position===child.position"> {{child.position}}</option>
-                    </select>
-                    <input style="width:55%" class="form-control"  ng-model="grpName" type="text" ng-class="{ 'has-error' : !isGroupNameValid(grpName)}" />
+                    <div layout-gt-xs="row">
+                    <md-input-container flex="35">
+                        <md-select placeholder="Select new position"
```
Comment (afrold) [inline igamt-lite-client/app/views/edit/editMessages.html]: @Jungyubw There should be an ng-show here ? 
`ng-show="editableGrp===node.obj.id"`

Comment (Jungyubw) [reply igamt-lite-client/app/views/edit/editMessages.html]: Now used ng-if.


On May 3, 2017, at 4:00 PM, Harold Affo <notifications@github.com<mailto:notifications@github.com>> wrote:


In igamt-lite-client/app/views/edit/editMessages.html<https://github.com/haffonist/igamt/pull/1681#discussion_r114638740>:

>                  <span ng-show="editableGrp===node.obj.id">
> -                    <select ng-show="editableGrp===node.obj.id" ng-model="position" ng-change="selectPos()"  style="width:15%" class="form-control input-sm">
> -                        <option ng-repeat="child in $root.segParent.children"  ng-selected="node.obj.position===child.position"> {{child.position}}</option>
> -                    </select>
> -                    <input style="width:55%" class="form-control"  ng-model="grpName" type="text" ng-class="{ 'has-error' : !isGroupNameValid(grpName)}" />
> +                    <div layout-gt-xs="row">
> +                    <md-input-container flex="35">
> +                        <md-select placeholder="Select new position"


@Jungyubw<https://github.com/Jungyubw> There should be an ng-show here ?
ng-show="editableGrp===node.obj.id"

File: igamt-lite-controller/src/main/java/gov/nist/healthcare/tools/hl7/v2/igamt/lite/web/config/Bootstrap.java [hunk 3]
```diff
@@ -275,12 +276,55 @@ public void afterPropertiesSet() throws Exception {
...
+	private void visitGroup(Group g, Message m) {
+		if (g.getName().contains(" ")) {
+			System.out.println("-------FOUND Group Name with Space----");
+			System.out.println(m.getScope());
+			System.out.println(g.getName());
+			System.out.println(g.getStatus());
+			System.out.println(g.getCreatedFrom());
+			System.out.println(g.getHl7Version());
+			String newGroupName = g.getName();
+			newGroupName = newGroupName.replaceAll(" ", "_");
```
Comment (afrold) [inline igamt-lite-controller/src/main/java/gov/nist/healthcare/tools/hl7/v2/igamt/lite/web/config/Bootstrap.java:316]: group name needs to be uppercase.

**Comment by Jungyubw** [comment]
In the our conversation, we didn’t decide it for existing data yet.
On May 3, 2017, at 4:08 PM, Harold Affo <notifications@github.com<mailto:notifications@github.com>> wrote:


@haffonist commented on this pull request.

________________________________

In igamt-lite-controller/src/main/java/gov/nist/healthcare/tools/hl7/v2/igamt/lite/web/config/Bootstrap.java<https://github.com/haffonist/igamt/pull/1681#discussion_r114640701>:

> +                             messageService.save(m);
+                       }
+               }
+
+       }
+
+       private void visitGroup(Group g, Message m) {
+               if (g.getName().contains(" ")) {
+                       System.out.println("-------FOUND Group Name with Space----");
+                       System.out.println(m.getScope());
+                       System.out.println(g.getName());
+                       System.out.println(g.getStatus());
+                       System.out.println(g.getCreatedFrom());
+                       System.out.println(g.getHl7Version());
+                       String newGroupName = g.getName();
+                       newGroupName = newGroupName.replaceAll(" ", "_");


group name needs to be uppercase.

File: igamt-lite-controller/src/main/java/gov/nist/healthcare/tools/hl7/v2/igamt/lite/web/config/Bootstrap.java (hunk 3 above)
Comment (afrold) [reply igamt-lite-controller/src/main/java/gov/nist/healthcare/tools/hl7/v2/igamt/lite/web/config/Bootstrap.java:316]:
```
We decided to enforce:
1.       No white space
2.       ALL CAPS letters
3.       Underscores
4.       Numbers
```

**Comment by Jungyubw** [comment]
So I just modified change space for existing data.

However, in the UI, we just allow Uppercase and number and DOT, and Underscore.


On May 3, 2017, at 4:09 PM, Woo, Jungyub (Assoc) <jungyub.woo@nist.gov<mailto:jungyub.woo@nist.gov>> wrote:

In the our conversation, we didn’t decide it for existing data yet.
On May 3, 2017, at 4:08 PM, Harold Affo <notifications@github.com<mailto:notifications@github.com>> wrote:


@haffonist commented on this pull request.

________________________________

In igamt-lite-controller/src/main/java/gov/nist/healthcare/tools/hl7/v2/igamt/lite/web/config/Bootstrap.java<https://github.com/haffonist/igamt/pull/1681#discussion_r114640701>:

> +                             messageService.save(m);
+                       }
+               }
+
+       }
+
+       private void visitGroup(Group g, Message m) {
+               if (g.getName().contains(" ")) {
+                       System.out.println("-------FOUND Group Name with Space----");
+                       System.out.println(m.getScope());
+                       System.out.println(g.getName());
+                       System.out.println(g.getStatus());
+                       System.out.println(g.getCreatedFrom());
+                       System.out.println(g.getHl7Version());
+                       String newGroupName = g.getName();
+                       newGroupName = newGroupName.replaceAll(" ", "_");


group name needs to be uppercase.

**Comment by afrold** [comment]
Why not ?


Harold Affo

--------------------------------------------------------------------
The contents of this e-mail and any attachments are intended solely for the use
of the named addressee(s) and may contain confidential and/or privileged
information. Any unauthorized use, copying, disclosure, or distribution of the
contents of this e-mail is strictly prohibited. If you are not the intended
recipient, please notify the sender immediately and delete this email. Thank you.


________________________________

**Review by afrold — CHANGES_REQUESTED** [review]

**Comment by Jungyubw** [comment]
Some HL7 based group has lowercase name.
Rob will ask about this to Frank and I am waiting for confirmation.


On May 3, 2017, at 4:12 PM, Harold Affo <notifications@github.com<mailto:notifications@github.com>> wrote:

Why not ?


Harold Affo

--------------------------------------------------------------------
The contents of this e-mail and any attachments are intended solely for the use
of the named addressee(s) and may contain confidential and/or privileged
information. Any unauthorized use, copying, disclosure, or distribution of the
contents of this e-mail is strictly prohibited. If you are not the intended
recipient, please notify the sender immediately and delete this email. Thank you.


________________________________

**Comment by afrold** [comment]
ok.


Harold Affo

--------------------------------------------------------------------
The contents of this e-mail and any attachments are intended solely for the use
of the named addressee(s) and may contain confidential and/or privileged
information. Any unauthorized use, copying, disclosure, or distribution of the
contents of this e-mail is strictly prohibited. If you are not the intended
recipient, please notify the sender immediately and delete this email. Thank you.


________________________________

**Pull request merged**